ToDoAppWithCoreData/Controllers/ListViewController.swift

```diff
@@ -0,0 +1,178 @@
+
+import UIKit
+import CoreData
+
+protocol ListViewControllerDelegate {
+    func tapSaveButton(item: String, detail: String, beginDate: Date, beginTime: Date, endDate: Date, endTime: Date, exitingItem: NSManagedObject?)
+}
+
+
+class ListViewController: UIViewController, UITextViewDelegate {
+    
+    public var item: String = ""
+    public var detail: String = ""
+    public var beginDate: Date!
+    public var beginTime: Date!
+    public var endDate: Date!
+    public var endTime: Date!
+    public var exitingItem: NSManagedObject?
+    
+    @IBOutlet var textFieldItem: UITextField!
+    @IBOutlet var detailTextView: UITextView!
+    @IBOutlet var saveButton: UIButton!
+    
+    let dateFormatter = DateFormatter()
+    let timeFormatter = DateFormatter()
+    
+    @IBOutlet var beginDatePicker: UIDatePicker!
+    @IBOutlet var beginTimePicker: UIDatePicker!
+    @IBOutlet var beginDateButton: UIButton!
+    @IBOutlet var beginTimeButton: UIButton!
+    @IBOutlet var endDatePicker: UIDatePicker!
+    @IBOutlet var endTimePicker: UIDatePicker!
+    @IBOutlet var endDateButton: UIButton!
+    @IBOutlet var endTimeButton: UIButton!
+    
+    var delegate: ListViewControllerDelegate?
+    
+    
+    override func viewDidLoad() {
+        
+        super.viewDidLoad()
+        
+        if(exitingItem != nil){
+            
+            textFieldItem.text = item
+            detailTextView.text = detail
+            beginDatePicker.date = beginDate!
+            beginTimePicker.date = beginTime!
+            endDatePicker.date = endDate!
+            endTimePicker.date = endTime!
+            
+        }
+        setDateAndTime()
+        setDetailTextView()
+    }
+    
+    
+    private func setDetailTextView() {
+        
+        detailTextView.layer.borderColor = UIColor.black.cgColor
+        detailTextView.layer.borderWidth = 1
+        detailTextView.layer.cornerRadius = 10
+        detailTextView.layer.masksToBounds = true
+        
+    }
+    
+    private func setDateAndTime() {
+        
+        dateFormatter.setLocalizedDateFormatFromTemplate("yMMdE")
+        timeFormatter.setLocalizedDateFormatFromTemplate("jm")
+        beginDateButton.setTitle("\(dateFormatter.string(from: beginDatePicker.date))", for: UIControl.State.normal)
```

Comment:
```suggestion
        beginDateButton.setTitle(dateFormatter.string(from: beginDatePicker.date), for: UIControl.State.normal)
```

ここ多分わざわざStringに埋め込む必要はないです